ide/models/project.py

```diff
@@ -158,11 +190,26 @@ def supported_platforms(self):
                 supported_platforms.append("diorite")
         return supported_platforms
 
+    @property
+    def resources_path(self):
+        return 'src/resources' if self.project_type == 'package' else 'resources'
+
+    @property
+    def is_standard_project_type(self):
+        return self.project_type in {'native', 'package'}
 
     def clean(self):
-        if isinstance(json.loads(self.app_keys), list) and self.sdk_version == "2":
+        is_sdk_2 = self.sdk_version == "2"
+        is_package = self.project_type == 'package'
+        if is_sdk_2 and self.uses_array_message_keys:
             raise ValidationError(_("SDK2 appKeys must be an object, not a list."))
-
+        if is_package:
+            if is_sdk_2:
+                raise ValidationError(_("Packages are not available for SDK 2"))
+            if not self.uses_array_message_keys:
+                raise ValidationError(_("Packages must use array based appmessage keys"))
```

Comment:
Interesting. The UI currently doesn't allow a user to use old-style appkeys in packages. Should I change that?